iptables/table.go

```diff
@@ -560,11 +596,60 @@ func (t *Table) getHashesFromBuffer(buf *bytes.Buffer) map[string][]string {
 	return newHashes
 }
 
-func (t *Table) InvalidateDataplaneCache() {
+func (t *Table) InvalidateDataplaneCache(reason string) {
+	log.WithField("reason", reason).Info("Invalidating dataplane cache")
 	t.inSyncWithDataPlane = false
 }
 
-func (t *Table) Apply() {
+var (
+	postUpdateRefreshDelays = []time.Duration{
+		1000 * time.Millisecond,
+		500 * time.Millisecond,
+		200 * time.Millisecond,
+		100 * time.Millisecond,
+		50 * time.Millisecond,
+	}
+)
+
+func (t *Table) Apply() (rescheduleAfter time.Duration) {
+	if t.inSyncWithDataPlane {
+		// We _think_ we're in sync, check if there are any reasons to think we might
+		// not be in sync.
```

Comment:
I'm thinking there must be a simpler alternative to this code - which is quite complex, and more or less a duplicate of that at the end of Apply.

Idea 1: can the IntDataplane invalidate before calling Apply, when it knows it's handling a resched kick?  But I think no, it can't, because IntDataplane schedules a kick for all tables at the minimum of the needed resched kick for all of them.  So if we did that, we'd be refreshing tables that didn't need it.

Idea 2: when it runs the code at the end of Apply, can the Table store off the time when it expects a refresh kick?  Then when Apply is called, if now >= that time, it invalidates.